lib/tasks/cleanup.rake

```diff
@@ -0,0 +1,20 @@
+# typed: false
+# frozen_string_literal: true
+
+namespace :cleanup do
+  desc 'Remove unattached files'
+  task uploads: :environment do
+    # Retrieve all blob keys that were created more than 7 days ago and are not attached
+    unattached_file_keys = ActiveStorage::Blob.where('created_at < ?', 7.days.ago)
+                                              .where.not(key: AttachedFile.all.map(&:key))
```

Comment:
This is going to run the system out of memory. The `AttachedFile.all.map(&:key)` part will grow without bound.  Can you figure out how to do this in a single query?

Reply:
I was worried about this, thanks for the feedback. I'll think about this.

Reply:
@jcoyne, working through this, I don't think a single query is possible (or more accurately, advisable). Since we need to base this on entries in the blob table that do not exist int he attachments table, we're going to face returning a lot of records. In this pass, I query for a single days worth of blob IDs and use that for the basis of the additional queries.

Totally open to suggestion or pairing if you have an idea of how to simplify.

Reply:
i think simply turning the inner `map(&:key)` into `select(:key)` might get you a single-shot query -- iirc, ActiveRecord is smart enough to use that inner `Relation` to build a single larger query, though here it can't do that here since the `map` gives a concrete list of already retrieved values.